### ISSUE TITLE
report-aggregate does not include report from current project

### ISSUE THREAD

**Issue by sidgate** [jacoco/jacoco#812]
### Steps to reproduce

Create a multi-module maven project with one module as aggregator. Following is a sample jacoco plugin configuration

```
<plugin>
                <groupId>org.jacoco</groupId>
                <artifactId>jacoco-maven-plugin</artifactId>
                <version>0.8.2</version>
                <executions>
                    <execution>
                        <id>prepare-agent</id>
                        <goals>
                            <goal>prepare-agent</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>report</id>
                        <phase>prepare-package</phase>
                        <goals>
                            <goal>report</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>report-aggregate</id>
                        <phase>verify</phase>
                        <goals>
                            <goal>report-aggregate</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

https://stackoverflow.com/questions/53983849/jacoco-report-aggregate-include-report-of-aggregate-project

JaCoCo version: 0.8.2
Operating system: Windows
Tool integration: Maven

### Expected behaviour
report-aggregate should include test report from current aggregate module

### Actual behaviour
report is generated only with dependent modules

The plugin does [load](https://github.com/jacoco/jacoco/blob/9c45dab3e9d9ed888481005095b30d3e5b55b338/jacoco-maven-plugin/src/org/jacoco/maven/ReportAggregateMojo.java#L109) the execution data for current project, but creates report only from [dependencies](https://github.com/jacoco/jacoco/blob/9c45dab3e9d9ed888481005095b30d3e5b55b338/jacoco-maven-plugin/src/org/jacoco/maven/ReportAggregateMojo.java#L135) . May we should just add following before the for loop

`support.processProject(group, getProject().getArtifactId(),
					getProject(), getIncludes(), getExcludes(), sourceEncoding);`

**Comment by Godin** [comment]
There are already two PRs #430 and #680 (so please search before creation of ticket in future). They are almost identical and contain your proposal, as well as description of why they was not accepted as is. In case if this description is unclear:

current behavior is there since the inception of `report-aggregate` and IMO [clearly stated in documentation](https://www.jacoco.org/jacoco/trunk/doc/report-aggregate-mojo.html)

> The report is created from all modules this project **depends on**.

proposed change will bother users who don't want this change - for example we don't need empty `org.jacoco.doc` in https://www.jacoco.org/jacoco/trunk/coverage/index.html , which is worse than addition of module for you.

Addition of configuration option to resolve this might require changes for existing users, and most importantly based on our experience unconditional clearly stated behavior is preferable than conditional. So maybe you can propose something better?

**Comment by sidgate** [comment]
Can we make it configurable, maybe `includeCurrentModule=true` ?

**Comment by gaellalire** [comment]
Just saw this bug, so I'm the third one to create a pull request #970 for the same thing with the same code. Sorry about this. The documentation may be clear but the behavior is not coherent. It's very disturbing that jacoco.exec of the current project is used but not its sources.

My use case is to test a multi module project, I want the coverage of all sources but for me it's OK if the test of one class is in another module. I just need to combine all jacoco.xml with diff-cover after. If you don't include the root project, I will have twice more jacoco.xml to combine.

I guess the `includeCurrentModule` boolean is a good solution.
Will you work on it ? Will you merge it if someone write the code for you ?

**Comment by ashu101** [comment]
please provide support for this. This is a blocker for us

**Comment by iNikem** [comment]
> current behavior is there since the inception of `report-aggregate` and IMO [clearly stated in documentation](https://www.jacoco.org/jacoco/trunk/doc/report-aggregate-mojo.html)
> 
> > The report is created from all modules this project **depends on**.
> 

Two sentences down from it is this:

> In addition execution data is collected from the project itself

Reading this I assume that the current project should be expected as well. At least that is how I read documentation.

**Comment by laradong** [comment]
How is it going on?

**Comment by paulo-santos** [comment]
> Add parameter to include the current project in the aggregated report #1007

This was implemented in version 0.8.9: https://github.com/jacoco/jacoco/pull/1007
You can use 
`<includeCurrentProject>true</includeCurrentProject>`

See https://www.jacoco.org/jacoco/trunk/doc/report-aggregate-mojo.html

**Comment by marchof** [comment]
@paulo-santos Thanks for pointing this out. I'll update this issue.
